test/integration/image-component/default/pages/on-loading-complete.js

```diff
@@ -1,74 +1,108 @@
+
+  return (
+    <div>
+      <h1>On Loading Complete Test</h1>
+      <ImageWithMessage
+        id="1"
+        src="/test.jpg"
+        layout="intrinsic"
+        width="128"
+        height="128"
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
+      />
+
+      <ImageWithMessage
+        id="2"
+        src={require('../public/test.png')}
+        placeholder="blur"
+        layout="fixed"
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
+      />
+
+      <ImageWithMessage
+        id="3"
+        src="/test.svg"
+        layout="responsive"
+        width="1200"
+        height="1200"
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
+      />
+
       <ImageWithMessage
         id="4"
         src="/test.ico"
         layout="fill"
         objectFit="contain"
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
       />
-    </div>
-    <hr />
-    <ImageWithMessage
-      id="5"
-      src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAMAAAAFCAYAAACAcVaiAAAAE0lEQVR42mNk+P+/ngEKGMngAADDdwx3Uz/3AAAAAABJRU5ErkJggg=="
-      layout="fixed"
-      width={64}
-      height={64}
-    />
-    <hr />
-    <div style={{ position: 'relative', width: '64px', height: '64px' }}>
+
       <ImageWithMessage
-        id="6"
+        id="5"
         src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAMAAAAFCAYAAACAcVaiAAAAE0lEQVR42mNk+P+/ngEKGMngAADDdwx3Uz/3AAAAAABJRU5ErkJggg=="
+        layout="fixed"
+        width={64}
+        height={64}
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
+      />
+
+      <ImageWithMessage
+        id="6"
+        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mNkYPhfDwAChwGA60e6kgAAAABJRU5ErkJggg=="
         layout="fill"
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
       />
+
+      <ImageWithMessage
+        id="7"
+        src="/test.bmp"
+        loading="eager"
+        width={400}
+        height={400}
+        idToCount={idToCount}
+        setIdToCount={setIdToCount}
+      />
+      <div id="footer" />
     </div>
-    <div id="footer" />
-  </div>
-)
+  )
+}
 
-function ImageWithMessage({ id, ...props }) {
+function ImageWithMessage({ id, idToCount, setIdToCount, ...props }) {
   const [msg, setMsg] = useState('[LOADING]')
+  const style =
+    props.layout === 'fill'
+      ? { position: 'relative', width: '64px', height: '64px' }
+      : {}
   return (
     <>
-      <Image
-        id={`img${id}`}
-        onLoadingComplete={({ naturalWidth, naturalHeight }) =>
-          setMsg(
-            `loaded img${id} with dimensions ${naturalWidth}x${naturalHeight}`
-          )
-        }
-        {...props}
-      />
+      <div className="wrap" style={style}>
+        <Image
+          id={`img${id}`}
+          onLoadingComplete={({ naturalWidth, naturalHeight }) => {
+            const count = idToCount[id] || 0
```

Comment:
```suggestion
            let count = idToCount[id] || 0
```

Reply:
Nice catch! No idea how this was working before 😬